source/js/RecipeCard.js

```diff
@@ -0,0 +1,251 @@
+      grid-template-columns: [left] auto [right];
+
+      background: #FFF6EC;
+    }
+
+    .thumbnail-photo {
+      height: 225px;
+      object-fit: cover;
+      width: 100%;
+    }
+
+    .rating-time {
+      display: grid;
+      grid-template-rows: [top] auto [bottom];
+      grid-template-columns: [left] 50% [middle] 50% [right];
+    }
+    `;
+    const styleElem = document.createElement('style');
+    styleElem.innerHTML=style;
+
+
+    const cleanData = {};
+    // process raw data into cleanData
+    cleanData.thumbnail = searchForKey(data,'thumbnailUrl');
+    cleanData.title = getRecipeTitle(data);
+    cleanData.url = getUrl(data);
+    cleanData.organization = getOrganization(data);
+    let cookTime = searchForKey(data,'cookTime');
+    let prepTime = searchForKey(data,'prepTime');
+    let totalTime = searchForKey(data,'totalTime');
+    cleanData.cookTime = convertTime(cookTime);
+    cleanData.prepTime = convertTime(prepTime);
+    cleanData.totalTime = convertTime(totalTime);
+    cleanData.ingredients = createIngredientList(searchForKey(data,'recipeIngredient'));
+    let tempRating = searchForKey(data,'aggregateRating');
+    if (!!tempRating) {
+      cleanData.rating = {
+        count: tempRating.ratingCount,
+        score: tempRating.ratingValue
+      }
+    }
+
+    const card = document.createElement('article');
+    card.classList.add('recipe-card');
+
+    const photo = document.createElement('img');
+    photo.classList.add("thumbnail-photo");
+    photo.setAttribute('src', cleanData.thumbnail);
+    card.appendChild(photo);
+
+    const title = document.createElement('p');
+    title.textContent = cleanData.title;
+    card.appendChild(title);
+
+    const info = document.createElement('div');
+    info.classList.add('rating-time');
+
+    const rating = document.createElement('p');
+    rating.textContent = `${cleanData.rating.score} stars`;
+    const starPicture = document.createElement('img');
+    starPicture.classList.add("star-image");
+    switch (Math.round(searchForKey(data, 'ratingValue'))) {
+      case 0:
+        starPicture.src = "images/0-star.svg";
+      case 1:
+        starPicture.src = "images/1-star.svg";
+      case 2:
+        starPicture.src = "images/2-star.svg";
+      case 3:
+        starPicture.src = "images/3-star.svg";
+      case 4:
+        starPicture.src = "images/4-star.svg";
+      case 5:
+        starPicture.src = "images/5-star.svg";
+    }
+    info.appendChild(rating);
+    info.appendChild(starPicture);
+
+    const time = document.createElement('p');
+    if (!!cleanData.prepTime) {
+      time.textContent = cleanData.prepTime;
+    } 
+    if (!!cleanData.cookTime) {
+      time.textContent = cleanData.cookTime;
+    }
+    if (!!cleanData.totalTime) {
+      time.textContent = cleanData.totalTime;
+    }
+    info.appendChild(time);
+
+    card.appendChild(info);
+
+    this.shadowRoot.appendChild(styleElem);
+    this.shadowRoot.appendChild(card);
```

Comment:
Codacy has a fix for the issue: [Prefer `Node#append()` over `Node#appendChild()`.](https://app.codacy.com/gh/cse110-fa21-group27/cse110-fa21-group27/pullRequest?prid=8447291)
```suggestion
    this.shadowRoot.append(card);
```